stingray/deadtime/fad.py

```diff
@@ -0,0 +1,189 @@
+
+    Other parameters
+    ----------------
+    plot : bool
+        Plot diagnostics
+    smoothing_alg : {'gauss', 'spline'}
+        Smoothing algorithm
+    smoothing_length : int
+        Number of bins to smooth in gaussian window smoothing
+    verbose: bool
+        Print out information on the outcome of the algorithm (recommended)
+    tolerance : float
+        Accepted relative error on the FAD-corrected Fourier amplitude, to be
+        used as success diagnostics.
+        Should be
+        ```
+        stdtheor = 2 / np.sqrt(n)
+        std = (average_corrected_fourier_diff / n).std()
+        np.abs((std - stdtheor) / stdtheor) < tolerance
+    strict : bool
+        Fail if condition on tolerance is not met.
+
+    Returns
+    -------
+    results : class:`astropy.Table` object
+        This table contains the results of the FAD correction, in its columns:
+        pds1: the corrected PDS of ``lc1``
+        pds2: the corrected PDS of ``lc2``
+        cs: the corrected cospectrum
+        ptot: the corrected PDS of lc1 + lc2
+
+    """
+    if smoothing_length is None:
+        smoothing_length = segment_size * 3
+    if gti is None:
+        gti = cross_two_gtis(lc1.gti, lc2.gti)
+    lc1.gti = gti
+    lc2.gti = gti
+    lc1.apply_gtis()
+    lc2.apply_gtis()
+    summed_lc = lc1 + lc2
+    start_inds, end_inds = \
+        bin_intervals_from_gtis(gti, segment_size, lc1.time,
+                                dt=lc1.dt)
+    freq = 0
+    pds1 = 0
+    pds2 = 0
+    ptot = 0
+    cs = 0
+    n = 0
+    average_diff = average_diff_uncorr = 0
+
+    if plot:
+        plt.figure()
+
+    for start_ind, end_ind in zip(start_inds, end_inds):
+        freq, f1, f1_leahy = _get_fourier_intv(lc1, start_ind, end_ind)
+        freq, f2, f2_leahy = _get_fourier_intv(lc2, start_ind, end_ind)
+        freq, ftot, ftot_leahy = \
+            _get_fourier_intv(summed_lc, start_ind, end_ind)
+
+        fourier_diff = f1_leahy - f2_leahy
+
+        if smoothing_alg == 'gauss':
+            smooth_real = gaussian_filter1d(fourier_diff.real ** 2,
+                                            smoothing_length)
+        else:
+            raise ValueError("Unknown smoothing algorithm: {}".format(
+                smoothing_alg))
+
+        if plot:
+            plt.scatter(freq, fourier_diff, s=1)
+
+        if all_leahy:
+            f1 = f1_leahy
+            f2 = f2_leahy
+            ftot = ftot_leahy
+        p1 = (f1 * f1.conj()).real
+        p1 = p1 / smooth_real * 2
+        p2 = (f2 * f2.conj()).real
+        p2 = p2 / smooth_real * 2
+        pt = (ftot * ftot.conj()).real
+        pt = pt / smooth_real * 2
+
+        c = (f2 * f1.conj()).real
+        c = c / smooth_real * 2
+
+        if n == 0 and plot:
+            plt.plot(freq, smooth_real, zorder=10, lw=3)
+            plt.plot(freq, f1_leahy, zorder=5, lw=1)
+            plt.plot(freq, f2_leahy, zorder=5, lw=1)
+
+        ptot += pt
```

Comment:
The length of the zeros depends on the fact that the number of light curve bins is even or odd (because of the selection of positive frequencies), so I think it would require an ugly piece of code with "ifs" that are bug-prone (what if numpy changes slightly the fft algorithm?) and not even needed for the final result. However, I added a comment to explain what will happen.